### ISSUE TITLE
eval loss not equal to task loss

### ISSUE THREAD

**Issue by rohanag** [deepset-ai/FARM#374]
**Question**
Hi, I'm finetuning a bert model using `RegressionProcessor` based off of the script `doc_regression.py`. During eval, I noticed loss output doesn't match MSE output. Shouldn't both of them be the same values? Loss for `RegressionProcessor` is MSE right? 

**Additional context**
Sample output 

Train ep

> och 4/5 (Cur. train loss: 0.0464): 100%|██████████████████████████████████████████████████████████████████████████| 3606/3606 [39:04<00:00,  1.90it/s]
> Evaluating: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 713/713 [02:22<00:00,  5.00it/s]
> 05/26/2020 19:47:29 - INFO - farm.eval -
> 
> \\|//       \\|//      \\|//       \\|//     \\|//
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ***************************************************
> ***** EVALUATION | TEST SET | AFTER 18030 BATCHES *****
> ***************************************************
> \\|//       \\|//      \\|//       \\|//     \\|//
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
> 05/26/2020 19:47:29 - INFO - farm.eval -
>  _________ regression _________
> 05/26/2020 19:47:30 - INFO - farm.eval -   loss: 0.12432620121493927
> 05/26/2020 19:47:30 - INFO - farm.eval -   task_name: regression
> 05/26/2020 19:47:30 - INFO - farm.eval -   mse: 0.11852640635928802
> 05/26/2020 19:47:31 - INFO - farm.eval -   report:
>  0.8790293516241948

Just to be clearer, the values 0.12432620121493927 and 0.11852640635928802 should match right?

**Comment by brandenchan** [comment]
Hi Rohan, I got a chance to look in to this finally. You're right in that both values are generated by MSE but the difference is that for loss, MSE is operating on raw logits, while the second value is calculated on rescaled logits. See [these lines here](https://github.com/deepset-ai/FARM/blob/6edf91e62637b589453775e138fea19aed171886/farm/modeling/prediction_head.py#L217). Hope that helps!

**Comment by rohanag** [comment]
thanks brandon!
